### PR TITLE
In-memory repos

### PR DESCRIPTION
**Pull request by vsalavatov**

### CODE REVIEW

File: internal/interface/repository/memory/articles.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	a.mutex.Lock()
+	defer a.mutex.Unlock()
+	offsetLeft := query.Offset
+	result := model.SearchResult{
+		TotalMatchesCount: 0,
+		Articles:          []model.ArticleMeta{},
+	}
+	processMatch := func(article model.Article) {
+		result.TotalMatchesCount++
+		if offsetLeft > 0 {
```
Comment (M0nsteR) [inline internal/interface/repository/memory/articles.go]: Could you make your logic look a bit clearer here?

File: internal/interface/repository/memory/articles.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+			offsetLeft--
+		} else if limit == 0 || uint32(len(result.Articles)) < limit {
+			result.Articles = append(result.Articles, article.ArticleMeta)
+		}
+	}
+	var re *regexp.Regexp
```
Comment (M0nsteR) [inline internal/interface/repository/memory/articles.go:70]: You need to think twice before entrusting your life to Regexp... Also, a separate function might be in order here.

File: internal/interface/repository/memory/subscriptions/article.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+		}
+		a.subscriptions[id] = append(subs, articleId)
+	}
+	return nil
+}
+
+func (a *ArticleSubscriptionRepo) UnsubscribeFromArticle(id model.UserId, articleId model.ArticleId) error {
+	a.mutex.Lock()
+	defer a.mutex.Unlock()
+	if subs, ok := a.subscriptions[id]; ok {
```
Comment (M0nsteR) [inline internal/interface/repository/memory/subscriptions/article.go]: Too many nested blocks, please make it clearer.

**Review by M0nsteR — CHANGES_REQUESTED** [review]
Please improve readability.

File: internal/interface/repository/memory/articles.go (hunk 2 above)
Comment (vsalavatov) [reply internal/interface/repository/memory/articles.go:70]: Regexp here is just a temporary solution until we will have a DB with full-text search capabilities

Comment (VasilyLupuleac) [reply internal/interface/repository/memory/articles.go:70]: Checking article for matches is now extracted into a separate function

File: internal/interface/repository/memory/subscriptions/article.go (hunk 3 above)
Comment (VasilyLupuleac) [reply internal/interface/repository/memory/subscriptions/article.go]: Unnecessary nested blocks are now removed and iterating over subscriptions is now moved to a separate function

File: internal/interface/repository/memory/articles.go (hunk 1 above)
Comment (VasilyLupuleac) [reply internal/interface/repository/memory/articles.go]: I have slightly improved readability

**Review by M0nsteR — APPROVED** [review]
Good to go.

**Pull request merged**